src/java.base/share/classes/java/lang/NamedPackage.java

```diff
@@ -86,4 +86,16 @@ URI location() {
     static Package toPackage(String name, Module module) {
         return new Package(name, module);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        return (o instanceof NamedPackage other)
+                && name.equals(other.name)
+                && module == other.module;
+    }
+
+    @Override
+    public int hashCode() {
+        return name.hashCode();
```

Comment:
I might be wrong, but IIRC object-identity-hash-code got/gets is treated in a special way w/ Lilliput, so I omitted `module` here.